### DISCUSSION TITLE
Is ARC_SUPPORT enabled in this source code?

### DISCUSSION THREAD

**Discussion by PhreakHeaven** [FormerLurker/ArcWelderPlugin#191] [Q&A]
My machine responds as follows:

**Send: M115**
Recv: FIRMWARE_NAME:Marlin Marlin 2.0.x (GitHub) SOURCE_CODE_URL:https://github.com/MarlinFirmware/Marlin PROTOCOL_VERSION:1.0 MACHINE_TYPE:3D Printer EXTRUDER_COUNT:2 UUID:cede2a2f-41a2-4748-9b12-c55c62f367ff
Recv: Cap:SERIAL_XON_XOFF:0
Recv: Cap:BINARY_FILE_TRANSFER:0
Recv: Cap:EEPROM:1
Recv: Cap:VOLUMETRIC:1
Recv: Cap:AUTOREPORT_TEMP:1
Recv: Cap:PROGRESS:0
Recv: Cap:PRINT_JOB:1
Recv: Cap:AUTOLEVEL:1
Recv: Cap:Z_PROBE:1
Recv: Cap:LEVELING_DATA:1
Recv: Cap:BUILD_PERCENT:0
Recv: Cap:SOFTWARE_POWER:0
Recv: Cap:TOGGLE_LIGHTS:0
Recv: Cap:CASE_LIGHT_BRIGHTNESS:0
Recv: Cap:EMERGENCY_PARSER:0
Recv: Cap:PROMPT_SUPPORT:0
Recv: Cap:AUTOREPORT_SD_STATUS:0
Recv: Cap:THERMAL_PROTECTION:1
Recv: Cap:MOTION_MODES:0
Recv: Cap:CHAMBER_TEMPERATURE:0
Recv: ok
Send: M155 S2
Recv: ok

-------------------------------------
AND
-------------------------------------

**Send: G3**
Recv: Error:G2/G3 bad parameters
Changing monitoring state from "Operational" to "Error: G2/G3 bad parameters"
Send: M112
Send: N2 M112*35
Send: N3 M104 T0 S0*34
Send: N4 M104 T1 S0*36
Send: N5 M140 S0*96
Changing monitoring state from "Error: G2/G3 bad parameters" to "Offline (Error: G2/G3 bad parameters)"
Connection closed, closing down monitor

So, this indicates to me that ARC_SUPPORT is NOT enabled. However, the source code provided by the manufacturer (Sovol 3D) for this machine (SV02), shows this in the 'Configuration.adv.h' file:

// @section extras

//
// G2/G3 Arc Support
//
#define ARC_SUPPORT               // Disable this feature to save ~3226 bytes
#if ENABLED(ARC_SUPPORT)
  #define MM_PER_ARC_SEGMENT  1   // Length of each arc segment
  #define MIN_ARC_SEGMENTS   24   // Minimum number of segments in a complete circle
  #define N_ARC_CORRECTION   25   // Number of interpolated segments between corrections
  //#define ARC_P_CIRCLES         // Enable the 'P' parameter to specify complete circles
  //#define CNC_WORKSPACE_PLANES  // Allow G2/G3 to operate in XY, ZX, or YZ planes
#endif

Am I missing something, or is ARC_SUPPORT actually enabled in the source code, but not the firmware provided by Sovol 3D? 

Additionally, is there a way I can utilize the customizations (configs) created for this machine with the newest Marlin (2.0.7.2) by simply copying the 'Configuration_adv.h' and 'Configuration_adv.h' files over and compiling (or something similar)? I'd like to flash the newest Marlin, and also retain the customization provided by Sovol 3D in the released source code, but I am currently unable to find the requisite information to properly achieve this.

Thank you for taking time to read, and assist, if possible!

**Comment by FormerLurker** [comment]
> Changing monitoring state from "Error: G2/G3 bad parameters" to "Offline (Error: G2/G3 bad parameters)"

This actually indicates arcs are enabled, so you should be good.  A G2/G3 without any parameters is a bad command, but if arcs weren't enabled your printer would have responded with ```Unknown Command```.

> flash the newest Marlin, and also retain the customization provided by Sovol 3D

You would have to manually merge in the changes from the latest marlin branch into the Sovol firmware.  That could be quite difficult.  I would recommend contacting the creators of the Sovol branch and asking them to bring their version up to date.  They may well be working on this.

I hope that helps!

**Reply by PhreakHeaven** [reply]
Thanks for the quick and informative response! I did contact Sovol already, and all they told me was "Sorry there is no news about that. And it's not recommended to update the firmware because we maybe unable to help with the problem if there was with the updated firmware."

I wonder what the most important changes to make would be, in order to match the machine, if I were to venture an attempt at modifying and compiling the newest Marlin firmware. I guess I'll just need to keep bugging them, because unfortunately the only configs available on GITHUB are for the SV01...but maybe I can adapt those to match the source code for the SV02?...probably not, but I don't know enough yet to determine this.

My problem right now with ARC WELDER is that I presumed ARC_SUPPORT was potentially disabled in the firmware (until I scanned the source code) because the nozzle kept slowing to a crawl (likely at the end of a path) and creating blobs. I did see you post a response to another user advising to use a beta, or dev version (?) of ARC WELDER instead, so maybe that's what I should try? Unless you have any other input to help me take advantage of your excellent work!

I really do prefer to update Marlin, since you recommend 2.0.6 at a minimum, but so far cannot find anything for guidance, as these machines appear to be pretty rare in the community, as far as support is concerned.

Thanks again!
